lib/proxy/validations.rb

```diff
@@ -14,8 +17,26 @@ def valid_mac? mac
 
   # validates the ip address
   def validate_ip ip
-    raise Error, "Invalid IP Address #{ip}" unless ip =~ /(\d{1,3}\.){3}\d{1,3}/
+    IPAddr.new(ip)
+    ip
+  rescue invalid_address_error
+    raise Error, "Invalid IP Address #{ip}"
+  end
+
+  # we may not want to raise error in some cases bu skip the ip instead
+  def soft_validate_ip ip
```

Comment:
That is a very confusing name. Can rely on "!" do signal presence of side-effects in calls. Then ```validate_ip``` becomes ```validate_ip!``` and ```soft_validate_ip``` becomes ```validate_ip```.